### PR TITLE
Launch tasks from admin UI

### PR DESCRIPTION
**Pull request by tonio**
Screen Shot 2020-05-05 at 11 22 35

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -231,6 +234,10 @@ import_contacts: ## Import contacts
 routes:
 	.build/venv/bin/proutes $(INI_FILE)
 
+.PHONY: tasks
+tasks:
+	docker-compose run --rm thinkhazard celery worker -A pyramid_celery.celery_app --ini $(INI_FILE)
+
```
Comment (arnaud-morvan) [inline Makefile:240]: This might not be useful anymore, if it is already in docker-compose.yaml

File: thinkhazard/celery.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+from thinkhazard.processing.decisiontree import DecisionMaker
+from thinkhazard.scripts.publish import main as publish_main
+imp = importlib.import_module("thinkhazard.processing.import")
+
+INI_FILE = os.environ["INI_FILE"]
+
+app = Celery()
+app.conf.broker_url = os.environ["BROKER_URL"]
+app.conf.result_backend = "s3"
+app.conf.s3_bucket = "thinkhazard"
```
Comment (arnaud-morvan) [inline thinkhazard/celery.py:18]: can be removed, see line 21

File: thinkhazard/celery.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+def publish():
+    print("start publish")
+    publish_main(argv=(None, INI_FILE.strip("'")))
+    print("end publish")
+
+
+@app.task
+def transifex_fetch():
+    print("start transifex_fetch")
+    time.sleep(10)
```
Comment (arnaud-morvan) [inline thinkhazard/celery.py:35]: Not finished, maybe add a TODO or FIXME

File: thinkhazard/celery.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    print("end admindivis")
+
+
+@app.task
+def process():
+    print("start processing")
+    Harvester.run((INI_FILE,))
+    Downloader.run((INI_FILE,))
+    Completer.run((INI_FILE,))
+    DecisionMaker.run((INI_FILE,))
```
Comment (arnaud-morvan) [inline thinkhazard/celery.py:52]: Missing Processor.run

File: thinkhazard/views/admin.py [hunk 5]
```diff
@@ -41,12 +39,39 @@
     HazardSet,
     Layer,
     TechnicalRecommendation,
+    Publication,
 )
+import thinkhazard.celery as tasks
 
+TASKS_LABELS = {
+    "publish": "Publication",
+    "transifex_fetch": "Import from Transifex",
```
Comment (arnaud-morvan) [inline thinkhazard/views/admin.py:48]: For info: we will also need "push to transifex" at sometime

File: thinkhazard/views/admin.py [hunk 6]
```diff
@@ -41,12 +39,39 @@
     HazardSet,
     Layer,
     TechnicalRecommendation,
+    Publication,
 )
+import thinkhazard.celery as tasks
 
+TASKS_LABELS = {
+    "publish": "Publication",
```
Comment (arnaud-morvan) [inline thinkhazard/views/admin.py:47]: Publish on public interface (start with verb to be consitent)

File: thinkhazard/views/admin.py [hunk 7]
```diff
@@ -41,12 +39,39 @@
...
+    "transifex_fetch": "Import from Transifex",
+    "admindivs": "Import administrative division from GeoNode",
+    "process": "Processing",
```
Comment (arnaud-morvan) [inline thinkhazard/views/admin.py:50]: Processing => Harvest and process Geonode datasets (start with verb to be consitent)

**Comment by arnaud-morvan** [comment]
When running publish, I get:
```
botocore.exceptions.ClientError: An error occurred (InvalidAccessKeyId) when calling the PutObject operation: The AWS Access Key Id you provided does not exist in our records.
```

**Comment by tkohr** [comment]
> When running publish, I get:
> 
> ```
> botocore.exceptions.ClientError: An error occurred (InvalidAccessKeyId) when calling the PutObject operation: The AWS Access Key Id you provided does not exist in our records.
> ```

On minio or AWS?

**Comment by tonio** [comment]
:clap: :clap: :clap:

**Pull request merged**